Client/Frontend/Browser/TabManager.swift

```diff
@@ -384,10 +384,26 @@ class TabManager: NSObject {
 
         let viableTabs: [Tab] = tab.isPrivate ? privateTabs : normalTabs
 
-        //If the last item was deleted then select the last tab. Otherwise the _selectedIndex is already correct
+        // Let's select the tab to be selected next.
         if let oldTab = oldSelectedTab, tab !== oldTab {
+            // If it wasn't the selected tab we removed, then keep it like that.
+            // It might have changed index, so we look it up again.
             _selectedIndex = tabs.index(of: oldTab) ?? -1
+        } else if let newTab = viableTabs.reduce(viableTabs.first, { currentBestTab, t2 in
+            if let t1 = currentBestTab, let time1 = t1.lastExecutedTime {
```

Comment:
nit: prefer tab1 over t1, because this code relates to time, where `t1` could be a time variable name.